data/core/macros/utils.cfg

```diff
@@ -578,6 +578,12 @@
             name=attack end
             delayed_variable_substitution=yes
 
+            [filter_condition]
+                [have_unit]
+                    id=$unit.id
+                [/have_unit]
+            [/filter_condition]
+
```

Comment:
Is the following chain of events possible? I'm basing this scenario on a failure mode @gfgtdf described to me the other day in a discussion about healing.

1. The `attack` event fires
2. The unit is stored and removed from play
3. This `attack end` does not fire, because its condition is false
4. The unit is unstored
5. The unit attacks, and has the FORCE_CHANCE_TO_HIT specials in effect even though it shouldn't be

Reply:
Sounds quite possible. Consider a "blink" ability which allows a Mage to "blink out of existence" for a moment, to avoid a hit. There may be other ways to code it, but the above sounds like a reasonable approach since it models the idea ... the unit simply isn't there. The down-side is having the unit come and go might cause other issues but it might be possible to work around those, as well.